dashboard_pianoroll.py

```diff
@@ -74,10 +74,16 @@ def main():
 
 
 def prepare_fragments(df: pd.DataFrame, piece: MidiPiece, n: int) -> list[dict]:
-    # Unpack
+    # Filter fragments with most variants (top 5)
+    df = df.nlargest(5, "n_variants")
+
     fragments = []
+    max_fragments = 5
+    max_variants = 5
```

Comment:
I think it would be better to separate this logic from the `prepare_fragments` function. If we want the dashboard to be helpful in reviewing different fragmentation methods, a structure like this would be better:

```python
fragments = prepare_fragments(df, piece, n)

filtering_method = st.select_box(
  label="Filtering method",
  options=["Top 5", "Fastes", "Longest"],
)

if filtering_method == "Top 5":
  filtered_fragments = top_five_filtering(fragments)
elif filtering_method == "Fastes":
  filtered_fragments = get_fastests_filtering(fragments)
...

for fragment in filtered_fragments:
  # Dsiplay
  ...
```

That way, you'll also be forced to come up with names for different filtering strategies, which will be useful in reviewing :)